### PR TITLE
sdk: remove Linux dependencies

### PR DESCRIPTION
**Pull request by elchead**
## Context

It is currently not possible to build the SDK / CLI on platforms other than Linux. One of the issues is that the imported packages include Linux specific code.

These components don't require it, so they shouldn't depend on it to allow for cross-platform usage.

[AB#4959](https://dev.azure.com/Edgeless/ae37573d-ccde-4af2-ab1e-001e587197d1/_workitems/edit/4959)

## Proposed changes
- move the issuer code into its own sub-package
- only let the components that require it (coordinator and initializer) import the issuer package

### CODE REVIEW

File: coordinator/main.go [hunk 1]
```diff
@@ -154,7 +155,7 @@ func newServerMetrics(reg *prometheus.Registry) *grpcprometheus.ServerMetrics {
 }
 
 func newGRPCServer(serverMetrics *grpcprometheus.ServerMetrics, log *slog.Logger) (*grpc.Server, error) {
-	issuer, err := atls.PlatformIssuer(log)
+	issuer, err := issuer.PlatformIssuer(log)
```
Comment (katexochen) [inline coordinator/main.go:158]: This is now stuttering.

File: internal/atls/issuer/issuer.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		return tdxissuer.New(
+			logger.NewWithAttrs(logger.NewNamed(log, "issuer"), map[string]string{"tee-type": "tdx"}),
+		), nil
+	default:
+		return nil, fmt.Errorf("unsupported platform: %T", cpuid.CPU)
+	}
+}
+
+// Issuer issues an attestation document.
+type Issuer interface {
```
Comment (burgerdev) [inline internal/atls/issuer/issuer.go:36]: The interface is also declared here: https://github.com/edgelesssys/contrast/blob/c75797e083533b282f3c19082e582e02dfd0bb82/internal/atls/atls.go#L88. I think we should only have one - here - which requires inlining the `Getter`.

File: internal/atls/issuer/issuer_linux.go [hunk 3]
```diff
@@ -1,22 +1,23 @@
+//go:build linux
+
 // Copyright 2024 Edgeless Systems GmbH
 // SPDX-License-Identifier: AGPL-3.0-only
```
Comment (burgerdev) [inline internal/atls/issuer/issuer_linux.go:4]:
```suggestion
// Copyright 2024 Edgeless Systems GmbH
// SPDX-License-Identifier: AGPL-3.0-only

//go:build linux
```

File: internal/atls/issuer/issuer_universal.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
+//go:build !linux
+
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
```
Comment (burgerdev) [inline internal/atls/issuer/issuer_universal.go:4]:
```suggestion
// Copyright 2024 Edgeless Systems GmbH
// SPDX-License-Identifier: AGPL-3.0-only

//go:build !linux
```

**Review by burgerdev — APPROVED** [review]
lgtm

**Pull request merged**